src/main/kotlin/com/fasterxml/jackson/module/kotlin/ConstructorValueCreator.kt

```diff
@@ -7,6 +7,6 @@ internal class ConstructorValueCreator<T>(override val callable: KFunction<T>) :
     override val accessible: Boolean = callable.isAccessible
 
     init {
-        callable.isAccessible = true
+        if (!accessible) callable.isAccessible = true
```

Comment:
Would you add a note here or in the class comment as to why `isAccessible` is being forced to true?

Reply:
Fixed.
5eca9cd